pandas/core/base.py

```diff
@@ -768,8 +769,21 @@ def base(self):
         return self.values.base
 
     @property
-    def _values(self):
-        """ the internal implementation """
+    def _ndarray_values(self):
+        """The data as an ndarray, possibly losing information.
+
+        The expectation is that this is cheap to compute.
+
+        - categorical -> codes
+
+        See '_values' for more.
+        """
+        # type: () -> np.ndarray
+        from pandas.core.dtypes.common import is_categorical_dtype
```

Comment:
this does NOT belong here. you *already* have a sub-clss EA for Categorical that can simply override this.

Reply:
This code is to return `_ndarray_values` for both Series and Index, and for Series there is in general no subclass that can override it. 
Of course we could say to put this logic in the Blocks, but that is then also not for Index, so not sure it is better.

This raises the question for me, though, what this will return for external extension types. Since it is is `.values`, I suppose this is a materialized array?

Reply:
I've moved it all the way to `ExtensionArray`, with the default being just returning an `np.array(self)`, same as before.  It's not part of the interface though. I think is what @jorisvandenbossche  suggested [here](https://github.com/pandas-dev/pandas/pull/19558/files#r167408753).